airflow/plugins/operators/littlepay_to_jsonl.py

```diff
@@ -0,0 +1,132 @@
+import csv
+import gzip
+import json
+import os
+from io import StringIO
+from typing import ClassVar, List
+
+import pendulum
+from calitp_data_infra.storage import (
+    PartitionedGCSArtifact,
+    fetch_all_in_partition,
+    get_fs,
+)
+from operators.littlepay_raw_sync import RawLittlepayFileExtract
+from tqdm import tqdm
+from tqdm.contrib.logging import logging_redirect_tqdm
+
+from airflow.models import BaseOperator
+
+LITTLEPAY_RAW_BUCKET = os.getenv("CALITP_BUCKET__LITTLEPAY_RAW")
+LITTLEPAY_PARSED_BUCKET = os.getenv("CALITP_BUCKET__LITTLEPAY_PARSED")
+
+
+class LittlepayFileJSONL(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = LITTLEPAY_PARSED_BUCKET
+    partition_names: ClassVar[List[str]] = ["instance", "extract_filename", "ts"]
+    extract: RawLittlepayFileExtract
+    filename: str
+
+    @property
+    def table(self) -> str:
+        return self.extract.table
+
+    @property
+    def instance(self) -> str:
+        return self.extract.instance
+
+    @property
+    def extract_filename(self) -> str:
+        return self.extract.filename
+
+    @property
+    def ts(self) -> pendulum.DateTime:
+        return self.extract.ts
+
+
+# TODO: outcome type; track unknown file types
+
+
+def parse_raw_file(file: RawLittlepayFileExtract, fs):
+    # mostly stolen from the Schedule job, we could probably abstract this
+    # assume this is PSV for now, but we could sniff the delimiter
+    filename, extension = os.path.splitext(file.filename)
+    assert extension == ".psv"
+    with fs.open(file.path) as f:
+        reader = csv.DictReader(
+            StringIO(f.read().decode("utf-8-sig")),
+            restkey="calitp_unknown_fields",
+            delimiter="|",
+        )
+    lines = [
+        {**row, "_line_number": line_number}
+        for line_number, row in enumerate(reader, start=1)
+    ]
+    jsonl_file = LittlepayFileJSONL(
+        extract=file,
+        filename=f"{filename}.jsonl.gz",
+    )
+    jsonl_file.save_content(
+        content=gzip.compress("\n".join(json.dumps(line) for line in lines).encode()),
+        fs=fs,
+    )
+
+
+class LittlepayToJSONL(BaseOperator):
+    template_fields = ()
+
+    def __init__(
+        self,
+        *args,
+        instance: str,
+        **kwargs,
+    ):
+        self.instance = instance
+        super().__init__(**kwargs)
+
+    def execute(self, context):
+        assert LITTLEPAY_RAW_BUCKET is not None and LITTLEPAY_PARSED_BUCKET is not None
+
+        fs = get_fs()
+
+        entities = [
+            "authorisations",
+            "customer-funding-source",
+            "device-transaction-purchases",
+            "device-transactions",
+            "micropayment-adjustments",
+            "micropayment-device-transactions",
+            "micropayments",
+            "product-data",
+            "refunds",
+            "settlements",
+        ]
+
+        with logging_redirect_tqdm():
+            for entity in tqdm(entities):
```

Comment:
It seems like there should be a way to run this on a chosen subset of entities if desired, but not necessary for initial implementation.

Reply:
I had thought about that but wasn't sure if 10 separate tasks was worth it right now, as our list of files is more static/known. For context @lauriemerrell created separate tasks per GTFS file in the Schedule pipeline since we only want to process a whitelist of known files from whatever's in zip files (RT does this as well for orchestration purposes). 

I'll add a TODO note.